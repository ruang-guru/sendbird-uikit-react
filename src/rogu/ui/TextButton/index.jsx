import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';
import { changeColorToClassName, Colors } from '../../../utils/color';

export default function TextButton({
  className,
  color,
  disabled,
  underline,
  onClick,
  children,
}) {
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        changeColorToClassName(color),
        'rogu-text-button',
        underline ? 'rogu-text-button--no-underline' : '',
        disabled ? 'rogu-text-button--disabled' : '',
      ].join(' ')}
      role="button"
      tabIndex={0}
      onClick={onClick}
      onKeyPress={onClick}
    >
      {children}
    </div>
  );
}

TextButton.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  color: PropTypes.string,
  disabled: PropTypes.bool,
  underline: PropTypes.bool,
  onClick: PropTypes.func,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
};

TextButton.defaultProps = {
  className: '',
  color: Colors.ONBACKGROUND_1,
  disabled: false,
  underline: false,
  onClick: () => {},
};
