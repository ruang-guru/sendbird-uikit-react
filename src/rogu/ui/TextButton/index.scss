@import '../../../styles/variables';

.rogu-text-button {
  cursor: pointer;
  display: inline-block;

  &.rogu-text-button--underline {
    text-decoration: underline;
  }

  &.rogu-text-button--disabled {
    cursor: not-allowed;

    @include themed() {
      color: t(on-bg-3);
    }
  }
}
