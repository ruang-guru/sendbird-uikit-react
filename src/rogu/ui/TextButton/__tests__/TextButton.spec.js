import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TextButton from '../index';

describe('TextButton', () => {
  it('should contain the className', function () {
    const className = 'example-classname';
    const component = shallow(
      <TextButton className={className}>Textbutton</TextButton>
    );

    expect(component.find('.rogu-text-button').hasClass(className)).toBe(true);
  });

  it('should do a snapshot test of the TextButton DOM', function () {
    const component = renderer.create(<TextButton>Textbutton</TextButton>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
