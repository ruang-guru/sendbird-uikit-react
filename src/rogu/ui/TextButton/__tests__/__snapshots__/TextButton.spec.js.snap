// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton should do a snapshot test of the TextButton DOM 1`] = `
<div
  className=" sendbird-color--onbackground-1 rogu-text-button  "
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  Textbutton
</div>
`;
